### PR TITLE
Add more llm guided metrics

### PR DESCRIPTION
**Pull request by bnativi**
Adds more llm guided metrics to the text generation PR. 

New metrics include: 
- bias (general text generation, llm-guided)
- context relevancy (RAG, llm-guided)
- faithfulness (RAG, llm-guided)
- hallucination (RAG, llm-guided)
- toxicity (general text generation, llm-guided)

The metrics have been added to the api and client. Functional, unit and integration tests have been written for all metrics. These metrics are also included in the example notebook. 

No major structural changes in this PR.

### CODE REVIEW

File: api/valor_api/backend/core/llm_clients.py [hunk 1]
```diff
@@ -26,6 +26,87 @@ class Messages(BaseModel):
...
+
+    Returns
+    -------
+    str
+        The instruction for the llm.
+    """
+    return f"""Based on the given text, please generate a comprehensive list of FACTUAL claims that can inferred from the provided text.
+
+**
+IMPORTANT: Please make sure to only return in JSON format, with the "claims" key as a list of strings. No words or explanation is needed.
```
Comment (ntlind) [inline api/valor_api/backend/core/llm_clients.py:46]: can these instructions be added to one of the docstrings? it looks like they are sitting outside of a function at the moment.

File: api/valor_api/backend/core/llm_clients.py [hunk 2]
```diff
@@ -26,6 +26,87 @@ class Messages(BaseModel):
...
+        The text to extract claims from.
+
+    Returns
+    -------
+    str
+        The instruction for the llm.
+    """
+    return f"""Based on the given text, please generate a comprehensive list of FACTUAL claims that can inferred from the provided text.
+
+**
```
Comment (ntlind) [inline api/valor_api/backend/core/llm_clients.py:45]: can all of this text be indented similarly?

File: docs/metrics.md [hunk 3]
```diff
@@ -310,20 +310,105 @@ print(detailed_evaluation)
 
 ## General Text Generation Metrics
 
+### Bias
+
+Bias is the proportion of biased opinions in a piece of text.
+
+Our implementation closely follows [DeepEval's implementation](https://github.com/confident-ai/deepeval/tree/main/deepeval/metrics/bias). We use the same two step prompting strategy and the same instructions. Notably, we follow the same [definitions of opinion and bias](https://docs.confident-ai.com/docs/metrics-bias#definition-of-bias).
+
+In deepeval, an opinion is defined according to the following principles:
```
Comment (ntlind) [inline docs/metrics.md:319]:
```suggestion
In DeepEval, an opinion is defined according to the following principles:
```

Comment (ntlind) [reply docs/metrics.md:319]: should we make this change everywhere for consistency?

File: docs/metrics.md [hunk 4]
```diff
@@ -310,20 +310,105 @@ print(detailed_evaluation)
...
+    - Biased: Crime rates are always higher in those big cities.
+    - Not Biased: Studies show a correlation between population density and certain types of crime.
+
 ### Coherence
 
 Coherence is a measure, on a scale of 1 to 5, of the collective quality of all sentences for a piece of text, with 5 indicating the highest coherence. The coherence of a piece of text is evaluated solely based on the text, without any reference to the query or any context. Because of this, the coherence metric can be applied to any text generation task.
 
 Valor's implementation of the coherence metric uses an instruction that was adapted from appendix A of DeepEval's paper G-EVAL: [NLG Evaluation using GPT-4 with Better Human Alignment](https://arxiv.org/pdf/2303.16634). While DeepEval's instruction and evaluation process was specific to summarization tasks, Valor generalized the instruction to apply to any text generation task. Most crucially, Valor does not use the datum text in its coherence evaluation.
 
+### Toxicity
```
Comment (ntlind) [inline docs/metrics.md:344]: nit: generally, these docs are really strong. I wish like metrics were grouped under some common header (e.g., toxicity and bias are similar in that they both deal with opinions)

File: api/valor_api/backend/core/llm_clients.py (hunk 2 above)
Comment (bnativi) [reply api/valor_api/backend/core/llm_clients.py:45]: Indenting would make the actual instruction indented. In DeepEval, they don't indent the instruction (for example https://github.com/confident-ai/deepeval/blob/main/deepeval/metrics/faithfulness/template.py).

File: docs/metrics.md (hunk 4 above)
Comment (bnativi) [reply docs/metrics.md:348]: There are already a lot of headers and subheaders. I wouldn't want to group these too much. Also I'm planning on moving coherence out of general text generation to make it summarization specific (as is how it is used elsewhere), so toxicity and bias might be our only two general text generation metrics.

**Pull request merged**